pithos/util.py

```diff
@@ -24,75 +24,128 @@
+                        [default_collection],
+                        None,
+                        on_unlock_finish,
+                        (service, default_collection),
+                    )
+
+                else:
+                    logging.debug('The default keyring is unlocked.')
+                    callback(None)
+
+        def on_get_finish(source, result, data):
+            try:
+                service = Secret.Service.get_finish(result)
+            except GLib.Error as e:
+                logging.error('Failed to get Secret.Service, Error: {}'.format(e))
+                callback(e)
+            else:
+                Secret.Collection.for_alias(
+                    service,
+                    Secret.COLLECTION_DEFAULT,
+                    Secret.CollectionFlags.NONE,
+                    None,
+                    on_for_alias_finish,
+                    service,
+                )
+
+        Secret.Service.get(
+            Secret.ServiceFlags.NONE,
+            None,
+            on_get_finish,
+            None,
         )
-        _current_collection = Secret.COLLECTION_SESSION
-        return
 
-    if not default_collection.get_locked():
-        logging.debug('The default keyring is unlocked.')
-    else:
-        num_items, unlocked = service.unlock_sync(
-            [default_collection],
+    def get_account_password(self, email, callback):
+        def on_password_lookup_finish(source, result, data):
+            try:
+                password = Secret.password_lookup_finish(result) or ''
+            except GLib.Error as e:
+                password = ''
+                logging.error('Failed to lookup password, Error: {}'.format(e))
+            callback(password)
+
+        Secret.password_lookup(
+            self._account_schema,
+            {'email': email},
+            None,
+            on_password_lookup_finish,
             None,
         )
 
-        if not num_items or default_collection not in unlocked:
-            _current_collection = Secret.COLLECTION_SESSION
-            logging.debug('The default keyring is locked. Using session collection.')
-        else:
-            logging.debug('The default keyring was unlocked.')
-
-
-def get_account_password(email):
-    return Secret.password_lookup_sync(_ACCOUNT_SCHEMA, {"email": email}, None) or ''
-
-
-def _clear_account_password(email):
-    return Secret.password_clear_sync(_ACCOUNT_SCHEMA, {"email": email}, None)
-
-
-def set_account_password(email, password, previous_email=None):
-    if previous_email and previous_email != email:
-        if not _clear_account_password(previous_email):
-            logging.warning('Failed to clear previous account')
-
-    if not password:
-        return _clear_account_password(email)
+    def set_account_password(self, email, password, callback):
+        def on_password_store_finish(source, result, data):
+            try:
+                success = Secret.password_store_finish(result)
```

Comment:
__[flake8]__
* [F841] local variable 'success' is assigned to but never used

<!-- Comment by [SideCI](https://sideci.com) -->